src/Datagrid/ProxyQuery.php

```diff
@@ -191,6 +191,14 @@ final public function isDistinct()
 
     public function execute(array $params = [], $hydrationMode = null)
     {
+        // NEXT_MAJOR: Remove this check and update method signature to `execute()`.
+        if ([] !== $params || null !== $hydrationMode) {
```

Comment:
We should use `func_num_args()` instead of checking the default values.
Even passing the default values should be deprecated.

Reply:
Addressed in #1334.